src/main/java/frc/robot/commands/BalanceCommand.java

```diff
@@ -0,0 +1,77 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.swerve.DriveConstants;
+import frc.robot.subsystems.Drivetrain;
+
+public class BalanceCommand extends CommandBase {
+
+    private Drivetrain m_drive;
+    private final PIDController m_pid;
+
+    private double m_currentAngle, m_output;
+
+    private boolean m_usePitch;
+
+    private int m_inverted;
+
+    Timer timer = new Timer();
+
+    public BalanceCommand(Drivetrain drive) {
+        m_drive = drive;
+        addRequirements(drive);
+
+        m_pid = new PIDController(DriveConstants.kBalanceP, DriveConstants.kBalanceI, DriveConstants.kBalanceD);
+        m_pid.setTolerance(DriveConstants.kBalanceTolerance);
+    }
+
+    @Override
+    public void initialize() {
+        m_inverted = 1;
+
+        m_pid.setSetpoint(0);
+        if(Math.abs(Math.PI/2 - Math.abs(m_drive.getAngleHeading())) > Math.PI/4)  { //Determines whether to use roll or pitch
+            m_usePitch = false;
+        }
+        else  {
+            m_usePitch = true;
+        }
+        if(m_drive.getAngleHeading() < 0)  {
+            m_inverted = -1;
+        }
```

Comment:
it seems m_inverted only factors in pitch, not roll. If the robot is sideways (using roll), then you should be checking if its absolute value is greater than pi/2